src/usr/bin/google_disk_expand

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+# Copyright 2020 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+root=$(grep -w / /proc/mounts|grep -v rootfs|cut -d' ' -f1)
+if [[ -z "$root" ]] || [[ ! -b "$root" ]]; then
+  echo "Couldn't determine root device"
+  exit 1
+fi
+
+echo "Checking if filesystem on ${root} needs resizing"
+
+if ! out=$(blkid -o udev "$root"); then
+  echo "Detecting fstype by blkid failed: ${out}"
+  exit 1
+fi
+  
+eval "$out"
+if [[ -z "$ID_FS_TYPE" ]]; then
+  echo "Unknown filesystem type"
+  exit 1
+fi
+
+case "$ID_FS_TYPE" in
+  ext*)
+    if ! out=$(resize2fs "$root"); then
+      echo "Calling resize2fs \"${root}\" failed: ${out}"
+      exit 1
+    fi
+    ;;
+  xfs*)
+    if ! out=$(xfs_growfs -d /); then
+      echo "Calling xfs_growfs -d / failed: ${out}"
+      exit 1
+    fi
+    ;;
```

Comment:
Add the case for all others... we should of course just exit but warn that it isn't a supported FS.